STYLE.md

```diff
@@ -131,15 +131,188 @@ if status is not pebble.ServiceStatus.ACTIVE:
+
+Example:
+
+- Avoid: "For reference information about the API, see [API and clients] and [API]." (What's the difference of these two?)
+- Prefer: "For an explanation of API access levels, see [API and clients]. For the full API reference, see [API reference]." (Clear)
+
+#### Don't over-promote
+
+Avoid overstatement: Only use adjectives like "comprehensive", "powerful", or "robust" when the feature truly meets the description.
+
+Example:
+
+- Avoid: "Pebble provides a comprehensive health check feature"
+- Prefer: "Pebble provides a health check feature"
+
+Let features speak for themselves, describe actual capabilities and use measurable terms when possible (e.g., "supports 3 types of checks" instead of "versatile checking")
+
+Future-proofing:
+
+- Avoid absolute terms unless permanently true
+- Qualify with "currently" when describing evolving features
+
+#### Avoid implementation-specific terminology
+
+Use generic descriptions instead of what's specific to the code. Focus on behaviour, not internal implementation: Describe what it does (e.g., "returns an error message") rather than how a particular implementation works.
+
+Example:
+
+- Avoid: "`Change.Err` will be non-empty if the change had an error." (Code-specific)
+- Prefer: "The `err` field in the response will contain an error message if the operation failed." (General)
+
+Exception: If documenting a client library (e.g., Go/Python SDKs), implementation details are appropriate.
+
+#### Articles
+
+"a" or "the", generic or specific, choose carefully. When describing generic parameters or behavior:
+
+- Avoid: "The format of the duration string is a sequence of decimal numbers"
+- Prefer: "The format of a duration string is a sequence of decimal numbers." (Describing a generic parameter, not a specific one.)
+
+- Avoid: "Restart the service when the health check fails."
+- Prefer: "Restart a service when the health check fails." (When describing a generic behaviour, no specific service is implied.)
+
+
+### Code blocks
+
+- Consistency: the styles of code blocks and terminal output samples should be the same, at least within the same document.
+- Preferred style: Use `{code-block}` when showing files. Use `{terminal}` (`.rst` style) when showing commands or terminal output. Avoid using `{code-block}` for commands or terminal output, unless it's consistent with the existing content in the same document.
+- Highlighting: Use `:emphasize-lines: 8-10` with `{code-block}` for highlighting lines in files when necessary. Don't add an inconsistent `{code-block}` so that you can highlight lines in commands or terminal output. Instead, use `{terminal}` along with helpful words and comments.
+
+### YAML
+
+- Types: Handle strings and integers carefully. For example, if the field "change" is a string in the code, use `change: "1"` in the YAML example, not `change: 1`.
```

Comment:
This is more of a code review point, not a style guide thing. I think we should drop it.